backend/danswer/db/document.py

```diff
@@ -46,13 +47,21 @@ def count_documents_by_needs_sync(session: Session) -> int:
     """Get the count of all documents where:
     1. last_modified is newer than last_synced
     2. last_synced is null (meaning we've never synced)
+    AND the document has a relationship with a connector/credential pair
+
+    TODO: The documents without a relationship with a connector/credential pair
+    should be cleaned up somehow eventually.
 
     This function executes the query and returns the count of
     documents matching the criteria."""
 
     count = (
-        session.query(func.count())
+        session.query(func.count(DbDocument.id.distinct()))
```

Comment:
was there a bug here?

Reply:
Essentially, we have documents that have no parent cc pair (because they only have permission sync data because the indexing hasnt run yet) so we dont want to try to sync those to vespa